php/system.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+$PI_HOLE_DIR = "/etc/pihole";
+
+$ENABLE_STUBS = true;
+
+if ($ENABLE_STUBS) {
+    $PI_HOLE_DIR = "stubs";
+}
+
+function isUsingStubs() {
+    global $PI_HOLE_DIR;
+    return $PI_HOLE_DIR == "stubs";
+}
+
+function getFileList($type) {
+    global $PI_HOLE_DIR;
+    return file_get_contents($PI_HOLE_DIR . "/" . $type . "list.txt");
```

Comment:
Ya, that should be sanitized, although you can't really get an `&` into it because that counts as adding another GET variable, although off the top of my head I can't think if the pipe would go through. Use [this](https://github.com/pi-hole/AdminLTE/blob/devel/php/queryads.php#L15-L21) to sanitize the domain.